lib/containers/common.pm

```diff
@@ -136,25 +138,27 @@ sub test_container_runtime {
 
 # Test a given image. Takes the image and container runtime (docker or podman) as arguments
 sub test_container_image {
-    my $image   = $_[0];
-    my $runtime = $_[1] //= "docker";
-    my ($name, $tag) = split(/:/, $image);
-    $tag //= 'latest';
+    my %args    = @_;
+    my $image   = $args{image};
+    my $runtime = $args{runtime};
+
+    die 'Argument $image not provided!'   unless $image;
+    die 'Argument $runtime not provided!' unless $runtime;
 
     # Pull the image if necessary
-    assert_script_run("$runtime pull $image", timeout => 300) if script_run("$runtime image ls | grep \"$image\" | grep \"$tag\"") != 0;
-    assert_script_run("$runtime image ls | grep '$name' | grep '$tag'");
-
-    my $container = "${runtime}_${name}_${tag}_smoketest";
-    $container =~ s!/!.!g;    # Slashes are not allowed as container names, but used for fetching images. Replace them with a dot
-    my $smoketest = "/bin/uname -r; /bin/echo \"Heartbeat from $name:$tag\"";
-    assert_script_run("$runtime container create --name '$container' '$name:$tag' /bin/sh -c '$smoketest'");
-    assert_script_run("$runtime container start '$container'");
-    assert_script_run("$runtime container logs '$container' > '/var/tmp/container_$container'");
-    assert_script_run("$runtime wait '$container'", 90);
-    assert_script_run("$runtime container rm '$container'");
-    assert_script_run("grep \"`uname -r`\" '/var/tmp/container_$container'");
-    assert_script_run("grep \"Heartbeat from $name:$tag\" '/var/tmp/container_$container'");
+    if (script_run("$runtime image inspect --format='{{.RepoTags}}' $image | grep '$image'") != 0) {
+        assert_script_run("$runtime pull $image", timeout => 300);
+        assert_script_run("$runtime image inspect --format='{{.RepoTags}}' $image | grep '$image'");
+    }
+
+    my $smoketest = "/bin/uname -r; /bin/echo \"Heartbeat from $image\"";
+    assert_script_run("$runtime container create --name 'testing' '$image' /bin/sh -c '$smoketest'");
+    assert_script_run("$runtime container start 'testing'");
+    assert_script_run("$runtime container logs 'testing' > '/var/tmp/container_testing'");
+    assert_script_run("$runtime wait 'testing'", 90);
```

Comment:
`$runtime wait 'testing'` needs to be before `$runtime container logs`, otherwise we have a race condition. Just swap lines 157 and 158